.github/workflows/ci.yml

```diff
@@ -359,19 +359,29 @@ jobs:
       - name: Upgrade ansible-core
         run: python3 -m pip install --upgrade ansible-core
 
-      - name: Download coverage artifacts
-        uses: actions/download-artifact@v3
+      - name: Download coverage artifacts A to H
+        uses: actions/download-artifact@v4
         with:
+          name: coverage-a-h
+          path: coverage
+
+      - name: Download coverage artifacts I to P
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage-i-p
+          path: coverage
+
+      - name: Download coverage artifacts Z to Z
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage-r-z0-9
           path: coverage
 
       - name: Combine coverage
         run: |
           make COLLECTION_VERSION=100.100.100-git install_collection
           mkdir -p ~/.ansible/collections/ansible_collections/awx/awx/tests/output/coverage
-          cd coverage
-          for i in coverage-*; do
-            cp -rv $i/* ~/.ansible/collections/ansible_collections/awx/awx/tests/output/coverage/
-          done
+          cp -rv coverage/* ~/.ansible/collections/ansible_collections/awx/awx/tests/output/coverage/
```

Comment:
this _feels_ equivalent

Reply:
It would be good to have better documentation. But this is tied into the change in `.github/actions/upload_awx_devel_logs/action.yml` in a non-obvious way. Because we previously used the same `name` for upload of each artifact. It would then put each item into a different folder based on the `path`. Using the same `name` is exactly what the v3-v4 change prohibited. So now we upload based on different names, but the `actions/download-artifact` changes here put them into the same local folder. So the loop is removed. Each item has kind of like coverage "fragments". Were were combining the fragments into the same folder eventually anyway, so yeah the ultimate behavior should be the same.

Reply:
> this feels equivalent

It's not quite. The previous one iterates all directories starting with `coverage-` and copies all their files into the target. The new one just copies all the `coverage-*` directories into the target. i.e. the first copies all the inner files of the directories, the second copies the directories themselves.

Something like this _might_ be equivalent if you really want to lose the loop:

`cp -rv coverage/coverage-*/* ~/.ansible/collections/ansible_collections/awx/awx/tests/output/coverage/`